src/controllers/googleAuthController.ts

```diff
@@ -0,0 +1,48 @@
+import User from '../models/User'
+import bcrypyt from 'bcrypt'
+import jwt from 'jsonwebtoken'
+
+class GoogleController {
+  static async googleAuth(req: any, res: any) {
+    try {
+      await User.findOne({
+        where: {
+          email: req.user.email,
+        },
+      }).then(async (result: any) => {
+        if (result) {
+          res.status(200).json({
+            success: true,
+            user: {
+              id: result.id,
+              firstName: result.firstName,
+              lastName: result.lastName,
+              email: result.email,
+              token: generateToken(result),
+            },
+          })
+        } else {
+          const salt = await bcrypyt.genSalt(10)
+          const hashpassword = await bcrypyt.hash(req.user.id, salt)
```

Comment:
a google authenticated user can also login using the userId as password  which is not safe because the user id is public.